app/src/sharedTest/java/org/oppia/app/home/ContinuePlayingActivityTest.kt

```diff
@@ -1,24 +1,304 @@
+          1,
+          R.id.story_name_text_view
+        )
+      ).check(
+        matches(
+          withText(containsString("Second Story"))
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testContinuePlayingActivity_recyclerViewItem1_topicNameIsCorrect() {
+    ActivityScenario.launch(ContinuePlayingActivity::class.java).use {
+      onView(withId(R.id.ongoing_story_recycler_view)).perform(
+        scrollToPosition<RecyclerView.ViewHolder>(
+          1
+        )
+      )
+      onView(
+        atPositionOnView(
+          R.id.ongoing_story_recycler_view,
+          1,
+          R.id.topic_name_text_view
+        )
+      ).check(
+        matches(
+          withText(containsString("First Topic"))
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testContinuePlayingActivity_recyclerViewItem1_lessonThumbnailIsCorrect() {
+    ActivityScenario.launch(ContinuePlayingActivity::class.java).use {
+      onView(withId(R.id.ongoing_story_recycler_view)).perform(
+        scrollToPosition<RecyclerView.ViewHolder>(
+          1
+        )
+      )
+      onView(
+        atPositionOnView(
+          R.id.ongoing_story_recycler_view,
+          1,
+          R.id.lesson_thumbnail
+        )
+      ).check(
+        matches(
+          withDrawable(R.drawable.lesson_thumbnail_graphic_duck_and_chicken)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testContinuePlayingActivity_recyclerViewItem3_showsLastMonthSectionTitle() {
+    ActivityScenario.launch(ContinuePlayingActivity::class.java).use {
+      onView(withId(R.id.ongoing_story_recycler_view)).perform(scrollToPosition<RecyclerView.ViewHolder>(3))
+      onView(
+        atPositionOnView(
+          R.id.ongoing_story_recycler_view,
+          3,
+          R.id.section_title_text_view
+        )
+      ).check(matches(withText(R.string.ongoing_story_last_month)))
+    }
+  }
+
+  @Test
+  fun testContinuePlayingActivity_recyclerViewItem3_showsSectionDivider() {
+    ActivityScenario.launch(ContinuePlayingActivity::class.java).use {
+      onView(withId(R.id.ongoing_story_recycler_view)).perform(scrollToPosition<RecyclerView.ViewHolder>(3))
+      onView(atPositionOnView(R.id.ongoing_story_recycler_view, 3, R.id.divider_view)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testContinuePlayingActivity_recyclerViewItem4_chapterNameIsCorrect() {
+    ActivityScenario.launch(ContinuePlayingActivity::class.java).use {
+      onView(withId(R.id.ongoing_story_recycler_view)).perform(
+        scrollToPosition<RecyclerView.ViewHolder>(
+          4
+        )
+      )
+      onView(
+        atPositionOnView(
+          R.id.ongoing_story_recycler_view,
+          4,
+          R.id.chapter_name_text_view
+        )
+      ).check(
+        matches(
+          withText(containsString("Miguel Reads a Book"))
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testContinuePlayingActivity_recyclerViewItem4_clickStoryItem_opensStoryActivityWithCorrectIntent() {
```

Comment:
So this is the sort of functionality we should test at the activity level (routing functionality), but we ought to test correctness at the fragment level. Can you move the other tests to a fragment-specific test suite that wraps the continue playing fragment in a test activity to verify functionality?

Reply:
@BenHenning Not able to understand this comment.

Reply:
Continuing from 1:1 discussion,

For now I have created a dummy `ContinuePlayingTestActivity` which inflates the `ContinuePlayingFragment`, also now the primary point for all test cases is `ContinuePlayingTestActivity` and not `ContinuePlayingActivity`.

Finally, I have divided test cases in two different files, `ContinuePlayingTestActivityTest` which tests only the routing cases and `ContinuePlayingFragmentTest` which tests everything related to `ContinuePlayingFragment`